client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js

```diff
@@ -1,152 +1,194 @@
-        return _.keys(self.agents.selected);
-    };
-
-    self.deleteAgents = function () {
-        self.agents.isDeleting = true;
-        return $q.all(self.getSelectedAgentIds().map(function (id) {
-            return OvhApiTelephony.OvhPabx().Hunting().Agent().v6().remove({
-                billingAccount: $stateParams.billingAccount,
-                serviceName: $stateParams.serviceName,
+    }
+
+    getSelectedAgentIds () {
+        return _.keys(this.agents.selected);
+    }
+
+    onTransformItemDone (items) {
+        this.$timeout(() => this.orderBy("number", this.orderedAscAgents));
+        return items;
+    }
+
+    deleteAgents () {
+        this.agents.isDeleting = true;
+        return this.$q.all(this.getSelectedAgentIds().map((id) =>
+            this.OvhApiTelephony.OvhPabx().Hunting().Agent().v6().remove({
+                billingAccount: this.$stateParams.billingAccount,
+                serviceName: this.$stateParams.serviceName,
                 agentId: id
-            }).$promise.then(function () {
-                _.pull(self.agents.ids, parseInt(id, 10));
+            }).$promise.then(() => {
+                _.pull(this.agents.id, parseInt(id, 10));
+            })))
+            .then(() => {
+                this.Toast.success(this.$translate.instant("telephony_alias_configuration_agents_delete_success"));
+                return this.fetchAgentsIds().then((ids) => {
+                    this.agents.ids = ids;
+                });
+            })
+            .catch((err) => {
+                this.Toast.error([this.$translate.instant("telephony_alias_configuration_agents_delete_error"), _.get(err, "data.message")].join(" "));
+                return this.$q.reject(err);
+            })
+            .finally(() => {
+                this.agents.isDeleting = false;
+                this.agents.selected = {};
             });
-        })).then(function () {
-            Toast.success($translate.instant("telephony_alias_configuration_agents_delete_success"));
-        }).catch(function (err) {
-            return new ToastError(err);
-        }).finally(function () {
-            self.agents.isDeleting = false;
-            self.agents.selected = {};
-        });
-    };
-
-    self.startEdition = function (agent) {
-        agent.inEdition = _.pick(agent, ["number", "simultaneousLines", "status", "timeout", "wrapUpTime"]);
-    };
-
-    self.isValidAgent = function (agent) {
-        var valid = true;
-        valid = valid && agent.number;
-        valid = valid && /^\d{1,6}$/.test(agent.timeout);
-        valid = valid && /^\d{1,6}$/.test(agent.wrapUpTime);
-        valid = valid && /^([1-9]|10)$/.test(agent.simultaneousLines);
+    }
+
+    startEdition (agent) {
+        agent.inEdition = _.pick(agent, ["description", "number", "simultaneousLines", "status", "timeout", "type", "wrapUpTime"]);
+    }
+
+    isValidAgent ({ number, timeout, simultaneousLines, wrapUpTime }) {
```

Comment:
Looks like a `static` method I guess. Isn't it?
ESLint task keeps silent due to: https://github.com/ovh-ux/ovh-manager-telecom/blob/master/.eslintrc.js#L21.
By removing this rule, I don't know you'll have a lots of errors :).